sdk/event_grid/src/event_grid_client.rs

```diff
@@ -0,0 +1,55 @@
+use crate::event_grid_request_builder::EventGridRequestBuilder;
+use crate::Event;
+use azure_core::errors::AzureError;
+use hyper::{self, client::HttpConnector, Method, StatusCode};
+use hyper_rustls::HttpsConnector;
+use serde::ser::Serialize;
+use url::Url;
+
+#[derive(Clone)]
+pub struct EventGridClient<'a, 'b> {
+    client: hyper::Client<HttpsConnector<HttpConnector>>,
+    pub topic_host_name: &'a str,
```

Comment:
I am not sure about borrowing here. It makes the client tightly coupled with the borrowed string, making it harder to reuse. 
In general, the clients tend to own their parameters so I would change it to String.